app/services/katello/registration_manager.rb

```diff
@@ -139,6 +140,8 @@ def unregister_host(host, options = {})
 
         if unregistering
           remove_host_artifacts(host)
+        elsif reregistering
+          remove_host_artifacts(host, kickstart_repository_id: host.content_facet.kickstart_repository_id)
```

Comment:
can never be too cautious :smile: 

```suggestion
          remove_host_artifacts(host, kickstart_repository_id: host.content_facet&.kickstart_repository_id)
```

Reply:
Should we place a logger.info or debug incase this happens again we can track down the params and host_artifacts for satellite installs?

Reply:
+1